### PR TITLE
Fix Win32 error messages on Unicode build

### LINKED ISSUE #1986: Truncated error message on Unicode build on Windows
When the generator plugin has errors, the printed error message is not helpful.
This was discovered on a Windows Chromium [build] (https://build.chromium.org/p/tryserver.chromium.win/builders/win_chromium_rel_ng/builds/274054) from this [CL](https://codereview.chromium.org/2034373002/#ps220001).

The error message is:
> [4190/24683] ACTION //third_party/dom_distiller_js:dom_distiller_proto_gen(//build/toolchain/win:x86)
> FAILED: pyproto/third_party/dom_distiller_js/dom_distiller_pb2.py gen/third_party/dom_distiller_js/dom_distiller.pb.cc gen/third_party/dom_distiller_js/dom_distiller.pb.h gen/third_party/dom_distiller_js/dom_distiller_json_converter.h 
> E:/b/depot_tools/python276_bin/python.exe ../../tools/protoc_wrapper/protoc_wrapper.py --proto-in-dir ../../third_party/dom_distiller_js/dist/proto --proto-in-file dom_distiller.proto --use-system-protobuf=0 -- ./protoc --python_out pyproto/third_party/dom_distiller_js --cpp_out gen/third_party/dom_distiller_js --plugin=protoc-gen-json_converter=../../third_party/dom_distiller_js/protoc_plugins/json_values_converter.py --json_converter_out=output_dir=:gen/third_party/dom_distiller_js
> --json_converter_out: protoc-gen-json_converter: %

Note that the message is just a single character.

### PR DESCRIPTION
**Pull request by wychen**
This fixes #1986.

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -159,6 +159,10 @@ else (MSVC)
   set(LIB_PREFIX)
 endif (MSVC)
 
+if (UNICODE)
+  add_definitions(-DUNICODE -D_UNICODE)
```
Comment (xfxyjwf) [inline cmake/CMakeLists.txt]: Drop this as it's no longer used?

Comment (wychen) [inline cmake/CMakeLists.txt]: It's still used. Without these lines, these macro won't be passed to the compiler, and defining UNICODE=ON has no effect.

Comment (xfxyjwf) [inline cmake/CMakeLists.txt]: Why do we need to pass UNICODE to the compiler (msvc in this case)? If the user want to enable unicode build, can they just enable it inside visual studio (the configuration page)?

Comment (wychen) [inline cmake/CMakeLists.txt]: It's for the test coverage on continuous integration. CommandLineInterfaceTest.Win32ErrorMessage can still pass on ANSI without the fix. In appveyor.yml, UNICODE=ON is added.

Comment (xfxyjwf) [inline cmake/CMakeLists.txt]: I see. Can you name the option the same way as other options? I.e., protobuf_UNICODE

Comment (wychen) [inline cmake/CMakeLists.txt]: Done.

**Comment by wychen** [comment]
Does it look good now?

**Comment by xfxyjwf** [comment]
LG. Thanks!

**Pull request merged**